File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
const solution = (board, moves) => {
  const stack = [];
  const LEN = board.length;
  let answer = 0;
  for (let i = 0; i < LEN; i++) {
    console.log(board[i]);
  }
  moves.forEach((element) => {
    for (let i = 0; i < LEN; i++) {
      const data = board[i][element - 1];
      if (data === 0) continue;
      if (data === stack[stack.length - 1]) {
        stack.pop();
        answer += 2;
      } else {
        stack.push(data);
      }
      board[i][element - 1] = 0;
      break;
    }
  });
  return answer;
};

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    int data;

    for (int move : moves) {
        for (int i = 0; i < board.size(); i++) {
            data = board[i][move - 1];
            if (data == 0) continue;
            if ((vector<int, allocator<int>>)data == board[board.size() - 1]) {
                board.pop_back();
                answer += 2;
            }
            else {
                board.push_back((vector<int, allocator<int>>)data);
            }
            board[i][move - 1] = 0;
            break;
        }
    };
    return answer;
}